### DISCUSSION TITLE
Why is `fresh.gen.ts` to be added to version control?

### DISCUSSION THREAD

**Discussion by retog** [freshframework/fresh#2394] [Q&A] [answered]
In my book generated files never go to version control.

**Comment by marvinhagemeister** [comment, marked as answer]
Before we introduce the build mode in Fresh a few months ago, everyone was using the automatic deployment mode in Deno Deploy. The way that works is that the contents of the GitHub repository is deployed directly without any additional steps in between. This means that there was no place in the process to generate files. That's where the recommendation to check `fresh.gen.ts` into version control comes from.

If you're using the build mode you should be able to exclude it. With the upcoming Fresh 2.0 release we went a step further and removed the need for having such a file entirely.
